### PR TITLE
Add Azure E2E tests

### PR DESCRIPTION
**Pull request by phillebaba**
This change adds a larger set of tests to Azure specific functions. The plan is that the tests should be run nightly against the main branch and use a custom subscription to run the tests in.

### CODE REVIEW

File: tests/azure/util_test.go [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	git2go "github.com/libgit2/git2go/v31"
```
Comment (stefanprodan) [inline tests/azure/util_test.go:12]: We can’t have this in flux CLI, please create a separate package with its own go.mod inside the test dir.

Comment (phillebaba) [reply tests/azure/util_test.go:12]: Done

File: go.mod [hunk 2]
```diff
@@ -3,6 +3,7 @@ module github.com/fluxcd/flux2
 go 1.16
 
 require (
+	github.com/Azure/go-autorest/autorest v0.11.18 // indirect
```
Comment (stefanprodan) [inline go.mod:6]: Can you undo these? I see no code changes in the CLI so I think we don't need any of these packages.

File: tests/azure/go.mod [hunk 3]
```diff
@@ -0,0 +1,106 @@
+module github.com/fluxcd/flux2/tests/azure
+
+go 1.17
```
Comment (stefanprodan) [inline tests/azure/go.mod:3]: Please use go 1.16 bumping the version to 1.17 should happen first in fluxcd/pkg then controllers and lastly flux2.

File: .github/workflows/e2e-azure.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+name: e2e-azure
+
+on:
+  workflow_dispatch:
```
Comment (stefanprodan) [inline .github/workflows/e2e-azure.yaml:4]: Can you please add here branch `azure/e2e` to see if the tests are passing?

Comment (phillebaba) [reply .github/workflows/e2e-azure.yaml:4]: Tests have been run fully without any issues.

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @phillebaba awesome work 🥇

**Pull request merged**